testing/trino-product-tests/src/main/java/io/trino/tests/product/iceberg/TestIcebergFormatVersionCompatibility.java

```diff
@@ -0,0 +1,67 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.tests.product.iceberg;
+
+import io.trino.tempto.ProductTest;
+import io.trino.tempto.assertions.QueryAssert;
+import org.testng.annotations.Test;
+
+import java.util.List;
+
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static io.trino.tempto.assertions.QueryAssert.Row.row;
+import static io.trino.tempto.assertions.QueryAssert.assertThat;
+import static io.trino.tests.product.TestGroups.ICEBERG_FORMAT_VERSION_COMPATIBILITY;
+import static io.trino.tests.product.TestGroups.PROFILE_SPECIFIC_TESTS;
+import static io.trino.tests.product.hive.util.TemporaryHiveTable.randomTableSuffix;
+import static io.trino.tests.product.utils.QueryExecutors.onCompatibilityTestServer;
+import static io.trino.tests.product.utils.QueryExecutors.onTrino;
+import static java.lang.String.format;
+import static org.testng.Assert.assertEquals;
+
+public class TestIcebergFormatVersionCompatibility
+        extends ProductTest
+{
+    private static final String TRINO_CATALOG = "iceberg";
+    private static final String TEST_SCHEMA_NAME = "default";
+
+    @Test(groups = {ICEBERG_FORMAT_VERSION_COMPATIBILITY, PROFILE_SPECIFIC_TESTS})
+    public void testTrinoTimeTravelReadTableCreatedByEarlyVersionTrino()
+    {
+        String baseTableName = "test_trino_time_travel_read_table_created_by_early_version_trino_" + randomTableSuffix();
+        String tableName = trinoTableName(baseTableName);
+
+        onCompatibilityTestServer().executeQuery(format("CREATE TABLE %s (c VARCHAR)", tableName));
+        onCompatibilityTestServer().executeQuery(format("INSERT INTO %s VALUES 'a', 'b', 'c';", tableName));
+
+        long latestSnapshotId = (long) onCompatibilityTestServer()
+                .executeQuery(format("SELECT snapshot_id FROM %s ORDER BY committed_at DESC FETCH FIRST 1 ROW WITH TIES", trinoTableName("\"" + baseTableName + "$snapshots\"")))
+                .getOnlyValue();
+        assertThat(onTrino().executeQuery(format("SELECT snapshot_id FROM %s ORDER BY committed_at DESC FETCH FIRST 1 ROW WITH TIES", trinoTableName("\"" + baseTableName + "$snapshots\""))))
+                .containsOnly(row(latestSnapshotId));
+
+        List<QueryAssert.Row> expected = onCompatibilityTestServer().executeQuery(format("SELECT * FROM %s", tableName)).rows().stream()
+                .map(row -> row(row.toArray()))
+                .collect(toImmutableList());
+        assertEquals(expected.size(), 3);
+        assertThat(onTrino().executeQuery(format("SELECT * FROM %s FOR VERSION AS OF %d", tableName, latestSnapshotId))).containsOnly(expected);
+
+        onCompatibilityTestServer().executeQuery(format("DROP TABLE %s", tableName));
+    }
+
+    private static String trinoTableName(String tableName)
```

Comment:
I saw similar method here `io.trino.tests.product.iceberg.TestIcebergSparkCompatibility#trinoTableName` maybe it makes sense to extract it to some utils ? Or make that one public and use here ?

Reply:
Yes, it's duplicated. Invoking method from the another test class is not a good idea, I think, because they are seperated part of testing and seems unrelated. Creating a method here is just for convenience and it's implemention is so simple. It seems no utility method exists to generate qualified table names. If using duplicated method here is not a good practice, I can directly use `String.format` just like other testing code.